radixdlt/src/integration/java/com/radixdlt/consensus/CrashFaultNetworkTest.java

```diff
@@ -17,52 +17,203 @@
+		List<Observable<Vertex>> committedObservables = allNodes.stream()
 			.map(bftNetwork::getVertexStore)
 			.map(VertexStore::lastCommittedVertex)
 			.collect(Collectors.toList());
 
 		Observable<Vertex> committed = Observable.zip(committedObservables, Arrays::stream)
-			.map(s -> s.distinct().collect(Collectors.toList()))
+			.map(committedVertices -> committedVertices.distinct().collect(Collectors.toList()))
 			.take(time, timeUnit)
 			.singleOrError()
-			.doAfterSuccess(s -> assertThat(s).hasSize(1))
-			.map(s -> (Vertex) s.get(0))
+			.doAfterSuccess(committedVertices -> assertThat(committedVertices).hasSize(1))
+			.map(vertices -> (Vertex) vertices.get(0))
 			.doAfterSuccess(v -> System.out.println("Committed " + v))
 			.toObservable();
 
 		bftNetwork.processBFT().takeUntil(committed)
 			.blockingSubscribe();
 	}
+
+	/**
+	 * Tests a static configuration of 4 nodes (1 crash-stopped), meaning a QC *can* be formed.
+	 * The intended behaviour is that the correct instances make progress,
+	 * "skipping" the faulty node as a leader when required.
+	 */
+	@Test
+	public void given_3_out_of_4_correct_bfts__then_correct_instances_should_get_same_commits_consecutive_vertices_over_1_minute() {
+		final int numNodes = 4;
+		final int numCrashed = 1;
+		final int numCorrect = numNodes - numCrashed;
+		final long time = 1;
+		final TimeUnit timeUnit = TimeUnit.MINUTES;
+
+		final List<ECKeyPair> correctNodes = createNodes(numCorrect, "c");
+		final List<ECKeyPair> faultyNodes = createNodes(numCrashed, "f");
+		final List<ECKeyPair> allNodes = Stream.concat(correctNodes.stream(), faultyNodes.stream()).collect(Collectors.toList());
+		final Set<ECPublicKey> correctNodesPubs = correctNodes.stream()
+			.map(ECKeyPair::getPublicKey)
+			.collect(Collectors.toSet());
+		final BFTTestNetwork bftNetwork = new BFTTestNetwork(allNodes);
+		// "crash" all faulty nodes by disallowing any communication
+		faultyNodes.forEach(node -> crashNode(node, bftNetwork));
+
+		// correct nodes should all get the same commits in the same order
+		Observable<Object> correctCommitCheck = Observable.zip(
+				correctNodes.stream()
+					.map(bftNetwork::getVertexStore)
+					.map(VertexStore::lastCommittedVertex)
+					.collect(Collectors.toList()),
+				Arrays::stream)
+			.map(committedVertices -> committedVertices.distinct().collect(Collectors.toList()))
+			.doOnNext(committedVertices -> assertThat(committedVertices).hasSize(1))
+			.map(vertices -> (Vertex) vertices.get(0))
+			.map(o -> o);
+
+		// correct nodes should only get timeouts when crashed nodes were a proposer
+		Observable<Object> correctTimeoutCheck = Observable.interval(1, TimeUnit.SECONDS)
+			.flatMapIterable(i -> correctNodes)
+			.doOnNext(cn -> assertThat(bftNetwork.getCounters(cn).getCount(Counters.CounterType.TIMEOUT))
+				.satisfies(new Condition<>(c -> c == (bftNetwork.getPacemaker(cn).getCurrentView().number() / numNodes) * numCrashed,
```

Comment:
Aren't there race conditions here?

Reply:
How so? The number of crashed nodes is static here

Reply:
if for example pacemaker timesout in between calls `getCount` and `getCurrentVew()`

Reply:
Ah yes, I figured the probability of that happening was super low. Do you think it matters?

Reply:
Don't think the probability of that is that low and false negatives here would just make things confuse people down the line. Can you think of a cleaner way of doing this check? Maybe a counter per node leader timeout instead of a global one?

Reply:
Possibly, I'll think about it.

Reply:
The simplest I've come up with is just checking that the number of actual timeouts is *smaller* than the maximum number of possible timeouts. In case of a race condition where the pacemaker times out between calls this still works.